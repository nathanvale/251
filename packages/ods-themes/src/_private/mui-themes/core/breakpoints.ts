import { Theme } from "@material-ui/core";
import { odsMasterTheme } from "../../odsMasterTheme";

function str2Num(bpValue?: string) {
  return Number((bpValue || "").split("px")[0]);
}

export const breakpoints: Partial<Theme["breakpoints"]> = {
  values: {
    xs: 0,
    sm: str2Num(odsMasterTheme.breakpoints.sm),
    md: str2Num(odsMasterTheme.breakpoints.md),
    lg: str2Num(odsMasterTheme.breakpoints.lg),
    xl: str2Num(odsMasterTheme.breakpoints.xl),
  },
};
