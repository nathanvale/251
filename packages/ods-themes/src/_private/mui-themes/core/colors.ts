import { odsMasterTheme } from "../../odsMasterTheme";

export const colors = {
  ...odsMasterTheme.colors,
};
