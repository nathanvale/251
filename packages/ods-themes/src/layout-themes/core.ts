import { Space, Theme, Transition } from "@origin-digital/ods-types";

const gridGutterWidth = 16;

const space: Space = [0, 4, 8, 12, 16, 24, 32, 48, 64];
space.none = space[0];
space.xxsmall = space[1];
space.xsmall = space[2];
space.small = space[3];
space.medium = space[4];
space.large = space[5];
space.xlarge = space[6];
space.xxlarge = space[7];
space.xxxlarge = space[8];

const transitions: Transition = {
  fast: "transform .15s ease, opacity .15s ease",
};

export const coreLayoutTheme: Theme = {
  space,
  gridGutterWidth,
  transitions,
  section: {
    maxWidth: {
      xs: 540,
      sm: 540,
      md: 720,
      lg: 960,
      xl: 1140,
    },
  },
};
