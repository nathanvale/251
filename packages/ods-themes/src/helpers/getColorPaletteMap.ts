import { ColorPaletteVariants } from "@origin-digital/ods-types";
import { Theme } from "@material-ui/core";

export function getColorPaletteMap(palette: Theme["palette"]) {
  const colorMap: Record<ColorPaletteVariants, string> = {
    white: palette.background.paper,
    disabled: palette.action.disabled,
    error: palette.error.main,
    errorDark: palette.error.dark,
    errorLight: palette.error.light,
    grey100: palette.grey[100],
    grey200: palette.grey[200],
    grey300: palette.grey[300],
    grey400: palette.grey[400],
    grey50: palette.grey[50],
    grey500: palette.grey[500],
    grey600: palette.grey[600],
    info: palette.info.main,
    infoDark: palette.info.dark,
    infoLight: palette.info.light,
    positive: palette.success.main,
    positiveDark: palette.success.dark,
    positiveLight: palette.success.light,
    primary: palette.primary.main,
    primaryB: palette.primaryB.main,
    primaryBDark: palette.primaryB.dark,
    primaryBLight: palette.primaryB.light,
    primaryDark: palette.primary.dark,
    primaryLight: palette.primary.light,
    promoteB: palette.promoteB.main,
    promoteBDark: palette.promoteB.dark,
    promoteBLight: palette.promoteB.light,
    promoteC: palette.promoteC.main,
    promoteCDark: palette.promoteC.dark,
    promoteCLight: palette.promoteC.light,
    secondary: palette.secondary.main,
    secondaryDark: palette.secondary.dark,
    secondaryLight: palette.secondary.light,
    secondaryB: palette.secondaryB.main,
    secondaryBDark: palette.secondaryB.dark,
    secondaryBLight: palette.secondaryB.light,
    success: palette.success.main,
    successDark: palette.success.dark,
    successLight: palette.success.light,
    warning: palette.warning.main,
    warningDark: palette.warning.dark,
    warningLight: palette.warning.light,
    transparent: "transparent",
    promote: palette.info.main,
    promoteLight: palette.info.light,
    promoteDark: palette.info.dark,
    //Tones
    critical: palette.primary.main,
    criticalLight: palette.primary.light,
    criticalDark: palette.primary.dark,
    caution: palette.secondary.main,
    cautionLight: palette.secondary.light,
    cautionDark: palette.secondary.dark,
    neutral: palette.grey[500],
    neutralLight: palette.grey[400],
    neutralDark: palette.grey[600],
  };
  return colorMap;
}
