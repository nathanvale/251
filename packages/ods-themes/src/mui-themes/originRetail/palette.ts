import { createMuiTheme, Theme } from "@material-ui/core";

export const palette: Theme["palette"] = createMuiTheme({
  palette: {
    primary: {
      main: "#ec0000",
    },
    secondary: {
      main: "#ffb432",
    },
    success: {
      main: "#008906",
    },
    info: {
      main: "#0a5fd2",
    },
    warning: {
      main: "#ffb432",
    },
    error: {
      main: "#ec0000",
    },
    action: {
      hoverOpacity: 0.2,
      disabled: "#e3e3e3",
    },
  },
}).palette;
