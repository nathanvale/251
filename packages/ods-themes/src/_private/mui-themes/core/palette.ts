import { createMuiTheme, Theme } from "@material-ui/core";
import { colors } from "./colors";

export const palette: Theme["palette"] = createMuiTheme({
  palette: {
    primary: {
      main: colors.red,
    },
    secondary: {
      main: colors.lightOrange,
    },
    success: {
      main: colors.green,
    },
    info: {
      main: colors.lightBlue,
    },
    warning: {
      main: colors.lightOrange,
    },
    error: {
      main: colors.red,
    },
    action: {
      hoverOpacity: 0.2,
      disabled: colors.grey16,
    },
    text: {
      primary: colors.grey,
      secondary: colors.grey80,
      disabled: colors.grey16,
    },
  },
}).palette;
