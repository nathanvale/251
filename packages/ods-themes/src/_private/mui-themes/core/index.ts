import createMuiTheme, {
  ThemeOptions,
} from "@material-ui/core/styles/createMuiTheme";

import { flattenPalette } from "../../helpers/flattenPalette";
import { palette } from "./palette";
import { breakpoints } from "./breakpoints";
import { shadows } from "./shadows";
import { getMUITypography } from "./typography";

export const themeOptions: ThemeOptions = {
  colors: flattenPalette(palette),
  palette,
  // Breakpoint
  breakpoints,
  // Shadows
  shadows,
  shape: {
    borderRadius: 0,
  },
  typography: getMUITypography(false),

  // Default props
  props: {
    MuiPaper: {
      square: true,
    },
    MuiTextField: {
      autoComplete: "false",
    },
  },

  // overrides: {
  //
  // }
};

export const coreMuiTheme = createMuiTheme(themeOptions);
