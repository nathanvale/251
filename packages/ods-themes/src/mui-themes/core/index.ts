import createMuiTheme, {
  ThemeOptions,
} from "@material-ui/core/styles/createMuiTheme";

import { palette } from "./palette";
import { breakpoints } from "./breakpoints";
import { shadows } from "./shadows";
import { getMUITypography } from "./typography";

export const themeOptions: ThemeOptions = {
  palette,
  // Breakpoint
  breakpoints,
  // Shadows
  shadows,
  shape: {
    borderRadius: 0,
  },
  typography: getMUITypography(false),

  // Default props
  props: {
    MuiPaper: {
      square: true,
    },
    MuiTextField: {
      autoComplete: "false",
    },
  },

  // overrides: {
  //
  // }
};

export const coreMuiTheme = createMuiTheme(themeOptions);
