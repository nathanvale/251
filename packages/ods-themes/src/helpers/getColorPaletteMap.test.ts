import { palette } from "../mui-themes/core/palette";
import { getColorPaletteMap } from "./getColorPaletteMap";

test("it should return a flattened MUI palette", () => {
  const colorMap = getColorPaletteMap(palette);
  expect(colorMap).toMatchSnapshot();
});
