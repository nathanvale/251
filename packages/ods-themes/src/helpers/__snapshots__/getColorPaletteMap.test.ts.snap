// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should return a flattened MUI palette 1`] = `
Object {
  "caution": "#ffb432",
  "cautionDark": "#b27d23",
  "cautionLight": "#ffc35b",
  "critical": "#ec0000",
  "criticalDark": "#a50000",
  "criticalLight": "#ef3333",
  "disabled": "#e3e3e3",
  "error": "#ec0000",
  "errorDark": "#a50000",
  "errorLight": "#ef3333",
  "grey100": "#f1f1f1",
  "grey200": "#e3e3e3",
  "grey300": "#ababab",
  "grey400": "#737373",
  "grey50": "#f8f8f8",
  "grey500": "#505050",
  "grey600": "#232323",
  "info": "#0a5fd2",
  "infoDark": "#074293",
  "infoLight": "#3b7fdb",
  "neutral": "#505050",
  "neutralDark": "#232323",
  "neutralLight": "#737373",
  "positive": "#008906",
  "positiveDark": "#005f04",
  "positiveLight": "#a5bb48",
  "primary": "#ec0000",
  "primaryB": "#fa4616",
  "primaryBDark": "#af310f",
  "primaryBLight": "#fb6b44",
  "primaryDark": "#a50000",
  "primaryLight": "#ef3333",
  "promote": "#0a5fd2",
  "promoteB": "#c34789",
  "promoteBDark": "#88315f",
  "promoteBLight": "#cf6ba0",
  "promoteDark": "#074293",
  "promoteLight": "#3b7fdb",
  "secondary": "#ffb432",
  "secondaryB": "#505050",
  "secondaryBDark": "#383838",
  "secondaryBLight": "#737373",
  "secondaryDark": "#b27d23",
  "secondaryLight": "#ffc35b",
  "success": "#008906",
  "successDark": "#005f04",
  "successLight": "#a5bb48",
  "transparent": "transparent",
  "warning": "#ffb432",
  "warningDark": "#b27d23",
  "warningLight": "#ffc35b",
  "white": "white",
}
`;
